### PR TITLE
feat: add terms violation error handling

### PR DESCRIPTION
**Pull request by nmrgt**
## Release Notes

- **New Features**
	- Enhanced database performance with new indexes on the `carpools` table.
	- Introduced new status `TermsViolationError` for carpool acquisitions.
	- Added functionality to verify terms violations during carpool requests.
	- New property `terms_violation_details` added to response objects for better clarity on compliance issues.
	- Expanded the `findByOperatorJourneyId` method to include terms violation labels.
	- Integrated new date manipulation functions for improved date handling in journey actions.
	- Updated API documentation to reflect new enumerations and schemas related to terms violations.

- **Bug Fixes**
	- Improved error handling for terms violations in journey registration.

- **Documentation**
	- Updated API specifications to include new terms violation labels and details.

- **Tests**
	- Added tests for terms violation handling and related functionalities.

### CODE REVIEW

File: api/src/pdc/providers/carpool/repositories/CarpoolLookupRepository.ts [hunk 1]
```diff
@@ -18,6 +18,67 @@ export class CarpoolLookupRepository {
...
+    if (operator_trip_id) {
+      filters.push(sql`cc.operator_trip_id <> ${operator_trip_id}`);
+    }
+
+    // operator_trip_id should be unique
+    // it may occurs some conflict between operators
+    // maybe use a composite count distinct
+    const sqlQuery = sql`
+      SELECT
+        count(distinct cc.operator_trip_id)
```
[the remark this answers is not in the document]
Comment (nmrgt) [reply api/src/pdc/providers/carpool/repositories/CarpoolLookupRepository.ts:70]: je me pose la question de la perf vs le risque de collision c'est pour ça qu'initialement je ne l'avais pas fait.
Un avis @jonathanfallon

File: doc/docs/.vuepress/public/specs/operateurs-api-v3.yaml [hunk 2]
```diff
@@ -700,6 +714,12 @@ paths:
                             temporal_overlap_duration_ratio:
                               type: number
                               description: ratio de temps de chevauchement entre les 2 trajets allant de 0 à 1. 1 désigne un chevauchement de temps total
+                  terms_violation_details:
+                    type: array
+                    description: | 
+                      Liste des raisons ayant abouti au rejet du trajet. Vide si pas de fraude détectée.
```
Comment (jonathanfallon) [inline doc/docs/.vuepress/public/specs/operateurs-api-v3.yaml:720]:
```suggestion
                      Liste des raisons ayant abouti au rejet du trajet.
```

Comment (jonathanfallon) [reply doc/docs/.vuepress/public/specs/operateurs-api-v3.yaml:720]: Ça n'est pas de la fraude, juste du non respect des CGU.

**Review by jonathanfallon — CHANGES_REQUESTED** [review]
Quelques modifs de texte dans la documentation

**Pull request merged**